lib/flagsmith.rb

```diff
@@ -113,6 +114,16 @@ def environment_data_polling_manager
     # You only need to call this if you wish to bypass environment_refresh_interval_seconds.
     def update_environment
       @_mutex.synchronize { @environment = environment_from_api }
+      update_identity_overrides
+    end
+
+    def update_identity_overrides
+      return unless @environment
+
+      @identity_overrides_by_identifier = {}
```

Comment:
I'm not sure why we need to instantiate this here, and in `initialize` ?

Reply:
In `initialize` to ensure that it's present even if it's accessed before `update_identity_overrides` is called and then again in `update_identity_overrides` to wipe out the stale data on successive runs.